tools/shared.py

```diff
@@ -2893,10 +2901,17 @@ def run_binaryen_command(tool, infile, outfile=None, args=[], debug=False, stdou
     if '--detect-features' not in cmd:
       cmd += Building.get_binaryen_feature_flags()
     print_compiler_stage(cmd)
+    # if we are emitting a source map, every time we load and save the wasm
+    # we must tell binaryen to update it
+    emit_source_map = Settings.DEBUG_LEVEL == 4 and outfile
+    if emit_source_map:
+      cmd += ['--input-source-map=' + infile + '.map']
+      cmd += ['--output-source-map=' + outfile + '.map']
+      cmd += ['--output-source-map-url=' + Settings.SOURCE_MAP_BASE + os.path.basename(Settings.WASM_BINARY_FILE) + '.map']
+
     if stdout is not None:
       return run_process(cmd, stdout=stdout).stdout
-    else:
-      run_process(cmd)
+    run_process(cmd)
```

Comment:
I think you can just always do `return run_process(cmd, stdout=stdout).stdout`.

Since `stdout=None` is the default and returning `.stdout` will just return None if redirection is active.